build/xtools.sh

```diff
@@ -56,6 +56,8 @@ MAKE_ARGS="TARGET_ARCH=${PRODUCT_ARCH} TARGET=${PRODUCT_TARGET}"
 MAKE_ARGS="${MAKE_ARGS} SRCCONF=${CONFIGDIR}/src.conf __MAKE_CONF="
 
 ${ENV_FILTER} make -C${SRCDIR} -j${CPUS} native-xtools ${MAKE_ARGS} NO_CLEAN=yes
+# XXX Temporary fix for cross build
+${ENV_FILTER} make -C${SRCDIR} -j${CPUS} native-xtools-install ${MAKE_ARGS} NO_CLEAN=yes
```

Comment:
looks like the proper fix since FBSD 12 introduced the new target and split the install stage from the old, maybe we can get away with changing native-xtools to native-xtools-install in the line above. Unfortunately that like is not backwards compatible so we have to be careful not breaking 11.x builds

Reply:
Actually this doesn't fix anything because nxb-bin was build on root folder (/nxb-bin) and not the actual OBJDIR. But for the sake of successful build, this is just a temporary fix. I found that many variables introduced by opnsense/tools doesn't retain it's data. And yes this is not backward compatible. only for 12.1.

Reply:
well, that is because on top of native-xtools-install they introduced DESTDIR so if it is empty everything expands to "/" root dir instead of the actual stage dir. Not sure about the other thing, but I'm going to check it out as soon as we have hardenedbsd patches in the tree again.

Reply:
20171028:
        The native-xtools target no longer installs the files by default to the 
        OBJDIR.  Use the native-xtools-install target with a DESTDIR to install 
        to ${DESTDIR}/${NXTP} where NXTP defaults to /nxb-bin.

Reply:
So that's the reason why some of these got empty string. Check out base and kernel too, I've marked them on my fork with `xxx temporary fix`. Maybe same thing happens on setup_stage().

And my current issue is that when running `make packages`, it complains about these:
```
make: "/etc/make.conf" line 0: 1 open conditional
make: Fatal errors encountered -- cannot continue
[: make:: unexpected operator
```

Reply:
should be 80e28f5e8dc53e

Reply:
Thanks, I'll test it now. Will let you know the results later.